CMakeLists.txt

```diff
@@ -254,131 +272,76 @@ endif()
 #   ARMADILLO_INCLUDE_DIRS - directories necessary for Armadillo includes
 #   BOOST_ROOT - root of Boost installation
 #   BOOST_INCLUDEDIR - include directory for Boost
+#   CEREAL_INCLUDE_DIR - include directory for cereal
 #   ENSMALLEN_INCLUDE_DIR - include directory for ensmallen
 #   STB_IMAGE_INCLUDE_DIR - include directory for STB image library
 #   MATHJAX_ROOT - root of MathJax installation
-find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
 
+if (CMAKE_CROSSCOMPILING)
+  get_deps(https://github.com/xianyi/OpenBLAS/releases/download/v0.3.13/OpenBLAS-0.3.13.tar.gz OpenBLAS OpenBLAS-0.3.13.tar.gz)
+  if (NOT MSVC)
+    execute_process(COMMAND make TARGET=ARMV8 BINARY=64 HOSTCC=gcc CC=${CMAKE_C_COMPILER} FC=${CMAKE_FORTRAN_COMPILER} NO_SHARED=1
+                    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/deps/OpenBLAS-0.3.13)
+    file(GLOB OPENBLAS_LIBRARIES "${CMAKE_BINARY_DIR}/deps/OpenBLAS-0.3.13/libopenblas.a")
+  endif()
+  find_library(GFORTRAN NAMES libgfortran.a PATHS ${CMAKE_SYSROOT}/usr/lib)
+  find_library(PTHREAD NAMES libpthread.a PATHS ${CMAKE_SYSROOT}/usr/lib)
+  set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${OPENBLAS_LIBRARIES} ${GFORTRAN} ${PTHREAD})
```

Comment:
I *think* that setting `OPENBLAS_LIBRARIES` here should be unnecessary---`find_package(Armadillo ...)` should return `ARMADILLO_LIBRARIES` with `libopenblas.a` in it.

Reply:
I will give a try locally and see what happen

Reply:
@rcurtin it did not work, `libopenblas` was not listed on the linker, I suppose that `ARMADILLO_LIBRARIES` was empty? If you think it is very important, I am happy to take sometime to see what is happening unless if it does not worth the effort.
Let me know what do you think

Reply:
No need to put a huge amount of time into it, but do you think you can add a `message(STATUS "Armadillo libraries: ${ARMADILLO_LIBRARIES}")` after the `find_package(Armadillo ...)` just so that we can see what it's finding?  I think you have things set up so that will run on Jenkins and we can both take a look at the log?

Reply:
Of course, I have added it, you can have a look, I have printed locally the variable is empty. Here is a link toward Jenkins:
http://ci.mlpack.org/job/aarch64/51/console

Reply:
I spent a long time trying to figure out what was going on here.  My digging took me into `CMake/FindArmadillo.cmake` where I found a few typos and bugs (I posted about that in the main review comment), and then into the CMake-supplied `FindBLAS.cmake` and `FindLAPACK.cmake`.  I discovered that we can get `FindBLAS.cmake` to find OpenBLAS when it's statically compiled, but only if we set these variables beforehand:

```
set(BLA_VENDOR OpenBLAS)
set(BLA_STATIC ON)
```

So, you could do that instead of `set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${OPENBLAS_LIBRARIES})`.

Next, I wanted to figure out why `ARMADILLO_LIBRARIES` did not end up with pthread or gfortran in it.  I couldn't figure this one out, because I could see that `find_package(Threads)` was being called from inside `FindBLAS.cmake`.  So, I can't come up with a better solution than what you are doing here.  But, I do think it would be better to add those two to `COMPILER_SUPPORT_LIBRARIES` instead:

```
  message(STATUS "Found gfortran: ${GFORTRAN}")
  message(STATUS "Found pthread: ${PTHREAD}")
  set(COMPILER_SUPPORT_LIBRARIES ${COMPILER_SUPPORT_LIBRARIES} ${GFORTRAN} ${PTHREAD})
```

Also, the `PATHS` is unnecessary (CMake will already search `${CMAKE_SYSROOT}/usr/lib/`):

```
  find_library(GFORTRAN NAMES libgfortran.a)
  find_library(PTHREAD NAMES libpthread.a)
```